ui_framework/src/components/form/field_search/field_search.js

```diff
@@ -3,28 +3,50 @@ import React, {
 } from 'react';
 import classNames from 'classnames';
 
+import {
+  KuiFormControlLayout,
+} from '../form_control_layout';
 
-export const KuiFieldSearch = ({ className, id, name, placeholder, value, ...rest }) => {
+import {
+  KuiValidatableControl,
+} from '../validatable_control';
+
+export const KuiFieldSearch = ({
+  className,
+  id,
+  name,
+  placeholder,
+  value,
+  isInvalid,
+  ...rest,
+}) => {
   const classes = classNames('kuiFieldSearch', className);
 
   return (
-    <input
-      type="search"
-      id={id}
-      name={name}
-      placeholder={placeholder}
-      className={classes}
-      value={value}
-      {...rest}
-    />
+    <KuiFormControlLayout
```

Comment:
This looks great. Other than the checkbox bit, this is OK by me to merge. The rest is just me theorizing.

---

So I'll just bring this up because it's something I struggled with. Originally I had these raw components include FormRow in them by default. It made it easy to pass props down.

You essentially went half way there by making form control layout and making it only control the icon and layout of the naked input.

My question I guess is... what's so different from this and just putting FormRow in here directly? That way people wouldn't even need to mess with form rows, the usage code is much cleaner. Basically the way I had it originally (before I changed it at your rec) was that you'd just say something like...

`<FieldSearch />` on its own.

This didn't pass labels, errors, invald states or icons. Icon was passed in the actual field_search.js as you're doing now through form_layout (before I just hadd it hardcoded into form_row in a similar forced fashion.

Essentially... do we need these things separated?

`<Fieldsearch />`

and

```
<FieldRow
  id="whatever"
  label="whatever"
>
  <Fieldsearch />
</FiedRow>
```

When we could just go ahead and put `<FieldRow>` inside of the component and just do this instead....

`<Fieldsearch/>` for naked.
`<Fieldsearch id="whatever" label="whatever" />` for one with labeling...etc.

Seems like a whole lot less to remember, and the HTML will print out just as clean. I guess I just don't see the reason why we need the two components living as separate, usable things.